pageserver/src/tenant/timeline.rs

```diff
@@ -2976,9 +2990,9 @@ impl Timeline {
             timeline = &*timeline_owned;
         }
 
-        if keyspace.total_size() != 0 {
-            return Err(GetVectoredError::MissingKey(keyspace.start().unwrap()));
-        }
+        // if keyspace.total_size() != 0 {
+        //     return Err(GetVectoredError::MissingKey(keyspace.start().unwrap()));
+        // }
```

Comment:
From our slack chat, the goal here is to remove the assumption that vectored get should operate only on existing key ranges (i.e. allow for gaps). One issue with that is that the search will always go all the way down to the root timeline.